ckan/views/user.py

```diff
@@ -672,6 +673,14 @@ def _prepare(self):
 
     def post(self) -> Response:
         self._prepare()
+
+        try:
+            captcha.check_recaptcha(request)
+        except captcha.CaptchaError:
+            error_msg = _(u'Bad Captcha. Please try again.')
+            h.flash_error(error_msg)
+            return h.redirect_to(u'/user/reset')
```

Comment:
Use `return h.redirect_to('user.request_reset')` instead of static path to be sure that URL is correct in edge cases.